src/ots.cc

```diff
@@ -233,6 +234,58 @@ bool ProcessTTF(ots::OpenTypeFile *header,
                         tables, file);
 }
 
+bool ProcessTTC(ots::OpenTypeFile *header,
+                ots::OTSStream *output, const uint8_t *data, size_t length) {
+  ots::Buffer file(data, length);
+
+  // we disallow all files > 1GB in size for sanity.
+  if (length > 1024 * 1024 * 1024) {
+    return OTS_FAILURE_MSG_HDR("file exceeds 1GB");
+  }
+
+  uint32_t ttc_tag;
+  if (!file.ReadTag(&ttc_tag)) {
+    return OTS_FAILURE_MSG_HDR("Error reading TTC tag");
+  }
+  if (ttc_tag != Tag("ttcf")) {
+    return OTS_FAILURE_MSG_HDR("Invalid TTC tag");
+  }
+
+  uint32_t ttc_version;
+  if (!file.ReadU32(&ttc_version)) {
+    return OTS_FAILURE_MSG_HDR("Error reading TTC version");
+  }
+  if (ttc_version != 0x00010000 && ttc_version != 0x00020000) {
+    return OTS_FAILURE_MSG_HDR("Invalid TTC version");
+  }
+
+  uint32_t num_fonts;
+  if (!file.ReadU32(&num_fonts)) {
+    return OTS_FAILURE_MSG_HDR("Error reading number of TTC fonts");
+  }
+  if (header->index >= num_fonts) {
+    return OTS_FAILURE_MSG_HDR("Requested font index is bigger than the number of fonts in the TTC file");
+  }
+
+  uint32_t offset_table_offsets[num_fonts];
```

Comment:
variable length arrays is a C99 feature which is not supported in Microsoft Visual C++ 2010, so build fails on Windows with the following error:

    src\ots.cc(270): error C2057: expected constant expression
    src\ots.cc(270): error C2466: cannot allocate an array of constant size 0
    src\ots.cc(270): error C2133: 'offset_table_offsets' : unknown size

Perhaps you could replace line 270 with something like this:

    uint32_t *offset_table_offsets = new uint32_t [num_fonts];

Cheers,
Cosimo